### PR TITLE
Bulk preserve ordering

### PR DESCRIPTION
**Pull request by kushagraThapar**
* Direct port of this PR: https://github.com/microsoft/kafka-connect-cosmosdb/pull/529

* Add a preserve ordering flag to leverage the preserve ordering bulk executor. The new flag is called `connect.cosmos.sink.bulk.ordering.preserved.enabled`

### CODE REVIEW

File: src/main/java/com/azure/cosmos/kafka/connect/CosmosDBConfig.java [hunk 1]
```diff
@@ -73,6 +73,9 @@ public class CosmosDBConfig extends AbstractConfig {
     public static final String COSMOS_SINK_BULK_COMPRESSION_ENABLED_CONF = "connect.cosmos.sink.bulk.compression.enabled";
     private static final String COSMOS_SINK_BULK_COMPRESSION_ENABLED_DOC = "Flag to indicate whether Cosmos DB in bulk mode will compress and resolve duplicates in the same batch to be written for Sink connector. By default it is true. ";
     private static final boolean DEFAULT_COSMOS_SINK_BULK_COMPRESSION_ENABLED = true;
+    private static final String COSMOS_SINK_BULK_ORDERING_PRESERVED_ENABLED_CONF = "connect.cosmos.sink.bulk.ordering.preserved.enabled";
```
Comment (xinlian12) [inline src/main/java/com/azure/cosmos/kafka/connect/CosmosDBConfig.java:76]: maybe "connect.cosmos.sink.bulk.preserveOrder.enabled"?

File: src/main/java/com/azure/cosmos/kafka/connect/CosmosDBConfig.java [hunk 2]
```diff
@@ -73,6 +73,9 @@ public class CosmosDBConfig extends AbstractConfig {
...
+    private static final String COSMOS_SINK_BULK_ORDERING_PRESERVED_ENABLED_DOC = "Flag to indicate whether Cosmos DB in bulk mode will preserve the ordering of items with the same id and partition key. By default it is true. ";
+    private static final boolean DEFAULT_COSMOS_SINK_BULK_ORDERING_PRESERVED_ENABLED = true;
```
Comment (xinlian12) [inline src/main/java/com/azure/cosmos/kafka/connect/CosmosDBConfig.java:78]: I think the default should be false -> probably will only change to true after split scenario being supported

**Review by xinlian12 — APPROVED** [review]
LGTM, thanks

File: src/main/java/com/azure/cosmos/kafka/connect/CosmosDBConfig.java (hunk 1 above)
Comment (kushagraThapar) [reply src/main/java/com/azure/cosmos/kafka/connect/CosmosDBConfig.java:76]: yeah, sounds good to me. lets change it.

**Comment by xinlian12** [comment]
Closing this PR as it is not applicable anymore.

**Pull request closed without merging**